### ISSUE TITLE
Usage of SQLPOINTER causing issues with SQLDriverConnectW

### ISSUE THREAD

**Issue by quinnj** [openlink/iODBC#50]
In the function signature [here](https://github.com/openlink/iODBC/blob/3f4d716839e2112d0ba87db2602edd88e15863c8/iodbc/connect.c#L2543), `szConnStrIn` has argument type `SQLPOINTER`, which seems to cause problems because of the difference in how `cbConnStrIn` is interpreted. In the [microsoft docs](https://docs.microsoft.com/en-us/sql/odbc/reference/syntax/sqldriverconnect-function?view=sql-server-ver15), it lists the arg type as SQLCHAR * (and for `W` SQLWCHAR *). Another example is the [MariaDB ODBC driver](https://github.com/mariadb-corporation/mariadb-connector-odbc/blob/0314e027ad3eda17d39469077d78e49f7e0d8977/odbc_3_api.c#L984) w/ the same SQLWCHAR * arg type.

So to summarize, I'm having trouble calling `SQLDriverConnectW` from my [wrapper library](https://github.com/JuliaDatabases/ODBC.jl/blob/cd5b9c51ea7ce38ec6155c6852ea3a7142dd9422/src/API.jl#L280) because if I pass the # of bytes (as expected by SQLPOINTER), then the driver complains, and if I pass the # of characters, iODBC doesn't get the full connection string.

Maybe I'm missing something on the iODBC side of things here? Or is it expected that applications just pass SQL_NTS?

**Comment by smalinin** [comment]
The function `SQLDriverConnect_Internal` is internal iODBC function, it isn't exported for using.
About `cbConnStrIn` looks the MS reference in your link above:
```
StringLength1
[Input] Length of *InConnectionString, in characters if the string is Unicode, or bytes if string is ANSI or DBCS.
```
For Unicode calls(SQLDriverConnectW) => `cbConnStrIn` is lengths in characters
For ANSI calls => it is lenght in bytes.

**Comment by quinnj** [comment]
Yes, I understand `SQLDriverConnect_Internal` is internal; I'm calling `SQLDriverConnectW` which in turn calls the internal method. The problem is that I'm saying I _do_ pass in the length in characters for `SQLDriverConnectW`, but [this line](https://github.com/openlink/iODBC/blob/3f4d716839e2112d0ba87db2602edd88e15863c8/iodbcinst/unicode.c#L2281) then divides my length _in characters_ by the size of `wchar_t`. So from my `lldb` debugging session, I was passing in a connection string like `DSN=ma` and the "converted" wide string was just `D`.

**Comment by smalinin** [comment]
Ok, I prepared fix for this, it will be commited soon.
This patch will be applied 
```
--- unicode.c.0	2020-05-29 15:27:56.722000000 +1000
+++ unicode.c	2020-05-29 15:13:25.428000000 +1000
@@ -2276,11 +2276,7 @@
   if (size == SQL_NTS)
     _wcxtoutf8 (charset, inStr, (char *)outStr, len);
   else
-    {
-      if (charset != CP_UTF8)
-        size /= _WCHARSIZE(charset);
-      _wcxntoutf8 (charset, inStr, (char *)outStr, size, len, NULL);
-    }
+    _wcxntoutf8 (charset, inStr, (char *)outStr, size, len, NULL);
 
   return outStr;
 }
```

**Comment by TallTed** [comment]
Fo ease of tracking, [committed on May 29](https://github.com/openlink/iODBC/commit/0a9f8d97abb74e4d380b00536b450169431a23aa).
